pkg/piecestore/rpc/server/server.go

```diff
@@ -53,24 +56,80 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (err error)
 	return server.Run(ctx)
 }
 
+//DiskStatus contains hard drive disk space info
+type DiskStatus struct {
+	All  uint64 `json:"all"`
+	Used uint64 `json:"used"`
+	Free uint64 `json:"free"`
+}
+
+// DiskUsage disk usage of path/disk
+func DiskUsage(path string) (disk DiskStatus) {
+	fs := syscall.Statfs_t{}
+	err := syscall.Statfs(path, &fs)
+	if err != nil {
+		return
+	}
+	disk.All = fs.Blocks * uint64(fs.Bsize)
+	disk.Free = fs.Bfree * uint64(fs.Bsize)
+	disk.Used = disk.All - disk.Free
+	return
+}
+
 // Server -- GRPC server meta data used in route calls
 type Server struct {
-	DataDir string
-	DB      *psdb.DB
-	pkey    crypto.PrivateKey
+	DataDir        string
+	DB             *psdb.DB
+	pkey           crypto.PrivateKey
+	totalAllocated uint64
 }
 
 // Initialize -- initializes a server struct
 func Initialize(ctx context.Context, config Config, pkey crypto.PrivateKey) (*Server, error) {
 	dbPath := filepath.Join(config.Path, "piecestore.db")
 	dataDir := filepath.Join(config.Path, "piece-store-data")
 
+	// read the allocated disk space from the config file
+	allocatedDiskSpace := config.AllocatedDiskSpace
+
+	// get the disk space details
+	diskSpace := DiskUsage("/")
+
 	db, err := psdb.Open(ctx, dataDir, dbPath)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Server{DataDir: dataDir, DB: db, pkey: pkey}, nil
+	// get how much is currently used, if for the first time totalUsed = 0
+	totalUsed, err := db.SumTTLSizes()
+	if err != nil {
+		//first time setup
+		totalUsed = 0x00
+	}
+
+	// check your hard drive is big enough
+	// on restarting the Piece node server, assuming already been working as a node
+	if uint64(totalUsed) != 0x00 {
```

Comment:
This `if` mountain is a little hard to follow. Flatten so all ifs are at the same level if possible to prevent weird logic for missed `if else` statements

Reply:
Flattened the if's